src/pages/index.js

```diff
@@ -1,62 +1,169 @@
 import Card from "../components/Card.js";
-// import { closePopup, openPopup } from "../utils/utils.js";
 import FormValidator from "../components/FormValidator.js";
 import Section from "../components/Section.js";
 import PopupWithForm from "../components/PopupWithForm.js";
+import PopupDeleteCard from "../components/PopupDeleteCard.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import UserInfo from "../components/UserInfo.js";
-import { initialCards, config } from "../utils/constants.js";
+import { config } from "../utils/constants.js";
+import Api from "../components/Api.js";
 import "./index.css";
+import { data } from "autoprefixer";
+
+export const api = new Api({
+  baseUrl: "https://around.nomoreparties.co/v1/group-12",
+  headers: {
+    authorization: "cd56466a-c69a-4082-86d8-efb553341f31",
+    "Content-Type": "application/json",
+  },
+});
 
 // profile buttons
 const profileAddBtn = document.querySelector("#profile-add-button");
 const profileEditBtn = document.querySelector("#profile-edit-btn");
+const profileImgEditBtn = document.querySelector("#edit-avatar-btn");
 // forms
 const editForm = document.querySelector("#profile-edit-modal");
 const addForm = document.querySelector("#profile-add-modal");
+const profileImageForm = document.querySelector("#edit-profile-img-modal");
 // form inputs
 const profileNameInput = editForm.querySelector("#profile-name-input");
 const profileDescriptionInput = editForm.querySelector(
   "#profile-description-input"
 );
 
-const userInfo = new UserInfo("#profile-name", "#profile-des");
+const userInfo = new UserInfo(
+  "#profile-name",
+  "#profile-des",
+  "#profile-image"
+);
+let userID;
+api
+  .getUserinfo()
+  .then((data) => {
+    const { name, about, avatar, _id } = data;
+    userID = _id;
+    userInfo.setUserInfo(name, about);
+    userInfo.setUserProfileImage(avatar);
+  })
+  .catch((error) => {
+    console.error("Error fetching data:", error);
+  });
 
 // functions
 function handleProfileFormSubmit(data) {
-  userInfo.setUserInfo(data);
-  profileEditPopup.close();
+  const { name, about } = data;
+  api
+    .updateProfile(name, about)
+    .then(() => {
+      profileEditPopup.editSubmitBtn();
+    })
+    .then(() => {
+      userInfo.setUserInfo(name, about);
+      profileEditPopup.close();
+      profileEditPopup.resetSubmitBtn();
```

Comment:
`      profileEditPopup.resetSubmitBtn();
`


You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case.


**Please, correct it everywhere**